components/public-api/gitpod/experimental/v1/workspaces.proto

```diff
@@ -28,6 +28,11 @@ service WorkspacesService {
     //   FAILED_PRECONDITION: if there's no running instance
     rpc StopWorkspace(StopWorkspaceRequest) returns (stream StopWorkspaceResponse) {}
 
+    // SendHeartbeat sends a heartbeat signal to a running workspace.
+    rpc SendHeartbeat(SendHeartbeatRequest) returns (SendHeartbeatResponse) {}
+
+    // SendCloseHeartbeat sends a close heartbeat signal to a running workspace.
+    rpc SendCloseHeartbeat(SendCloseHeartbeatRequest) returns (SendCloseHeartbeatResponse) {}
```

Comment:
FWIW, I'm not convinced these should exist on the Public API, at least not on the Workspace Service. Why? Because these are not directly usable by our users, and are mostly an implementation detail of how we communicate with the rest of the system.

I think these are a good use-case for a private API package which still benefits from the API quality aspects, but doesn't actually get exposed to the public users. 

For now, we can keep them here but when we move the APIs to the "stable" package, I wouldn't include them.
Maybe we can for now move them under a `WorkspaceInstance` service, or something closer related to IDE behaviour, rather than a Workspace behaviour and properties. What do you think?

Reply:
For now we can move it to some experimental service like IDEWorkspaceInstanceService.

In future we would like to allow 3rd party IDEs, instead of dealing with supporting each of it. It is part of our long term motivation in Public API then integrators should be able to report heartbeating and closing.

Reply:
Got it. Perhaps `WorkspaceIDEService` makes sense for that use-case

Reply:
In general, I'd like each service to be as minimal as possible, with additional services possibly layering on top and composing the basic services. Here, I'd see it as `IDEWorkspaceService` -> `WorkspaceService.GetWorkspace` -> use that info to do something to the instance for example